lib/events/complete.go

```diff
@@ -303,3 +314,38 @@ loop:
 	}
 	return nil
 }
+
+type MockSessionTrackerService struct {
```

Comment:
This looks like test only code.

- If we can unexport it, we should. (though it looks like you're using it in multiple packages)
- If we can move it to an _test.go file, we should.
- If we can't do either of the above, can we move it to `lib/events/eventstest`?